tests/unit/general/test_mem.cpp

```diff
@@ -12,11 +12,69 @@
 #include "mfem.hpp"
 #include "unit_tests.hpp"
 
+using namespace mfem;
+
+TEST_CASE("MemoryManager/Scopes",
+          "[MemoryManager]"
+          "[CUDA]")
+{
+   SECTION("WithNewMemoryAndSize")
+   {
+      Vector x(1);
+      x.UseDevice(true);
+      {
+         Vector X;
+         // from Operator::InitTVectors
+         X.NewMemoryAndSize(x.GetMemory(), x.Size(), false);
+         // from Vector::SetSubVectorComplement
+         X.Read();
+         // from Operator::RecoverFEMSolution
+         x.SyncMemory(X);
+      }
+      // Accessible Memory<double> to get the flags
+      struct MemoryDouble
+      {
+         double *h_ptr;
+         int capacity; ///< Size of the allocated memory
+         MemoryType h_mt; ///< Host memory type
+         mutable unsigned flags;
+      };
+      const MemoryDouble *mem = (MemoryDouble*) &x.GetMemory();
+      const double *h_x = mem->h_ptr;
+      REQUIRE(h_x == x.GetData());
+      REQUIRE(mem->capacity == x.Size());
+      REQUIRE(mem->h_mt == Device::GetHostMemoryType());
+      constexpr unsigned REGISTERED = 1 << 0;
+      const bool registered = mem->flags & REGISTERED;
+      const bool registered_is_known = registered == mm.IsKnown(h_x);
+      REQUIRE(registered_is_known);
+   }
+
+   SECTION("WithMakeRef")
+   {
+      Vector x(1);
+      x.UseDevice(true);
+      const double *x_data = x.GetData();
+      {
+         Vector X;
+         // from Operator::InitTVectors
+         X.MakeRef(x, 0, x.Size());
+         // from Vector::SetSubVectorComplement
+         X.Read();
+         // from Operator::RecoverFEMSolution
+         x.SyncMemory(X);
+      }
+      REQUIRE((x_data == x.HostRead()));
+   }
+}
+
+// Debug device specific tests, not done on Windows
 #ifndef _WIN32
 #include <unistd.h>
 
-using namespace mfem;
-
+// FIXME: re-enable this test when fixed to reset, not destroy the global
+//        memory manager, after it is done.
+#if 0
```

Comment:
We should move it as a stand alone test if the tweaks of this PR introduce new requirements.

Reply:
I agree, I think we should move it.